### ISSUE TITLE
Scoreboard handler is NULL

### ISSUE THREAD

**Issue by multicomunicaec** [Deltik/mod_antiloris#3]
Hi, i use mod_antiloris in my apache servers with versions: 

- Server version: Apache/2.4.29 (Ubuntu), Server built:   2023-03-08T17:34:33, Server's Module Magic Number: 20120211:68
- Server version: Apache/2.4.58 (Ubuntu), Server built:   2024-07-17T18:55:23, Server's Module Magic Number: 20120211:129

I have a problem when make the request to the servers.

First of all, i redirect all http traffic to https, then i have the next issue:

for some reason, (maybe by the redirect) when I curl my server, the first connection is corrected process by mod_antiloris, but inmmediatly enter a second connection, but in this connection the sbh is NULL, then crash my server.

`sb_handle *sbh = c->sbh;`
https://github.com/Deltik/mod_antiloris/blob/main/mod_antiloris.c#L293

Why this happend, and how I can resolve this?

**Comment by Deltik** [comment]
I see that it is possible for the scoreboard handle to be NULL [when a connection is suspended in the event MPM](https://github.com/apache/httpd/blob/c72b74ccef72fb56becd55888a3aa783c4ab5b6b/server/mpm/event/event.c#L877).  I also see that throughout the scoreboard code (like [here](https://github.com/apache/httpd/blob/c72b74ccef72fb56becd55888a3aa783c4ab5b6b/server/scoreboard.c#L637-L638)), there are NULL checks on the scoreboard handle.

This core piece of code is over 15 years old (from the very beginning of mod_antiloris), and it surprises me that [the hook named `pre_connection()`](https://github.com/Deltik/mod_antiloris/blob/018eed86d30e38a0a8ffe05db6f76c2f1f44cc4c/mod_antiloris.c#L98) is [actually](https://github.com/Deltik/mod_antiloris/blob/018eed86d30e38a0a8ffe05db6f76c2f1f44cc4c/mod_antiloris.c#L143) a [`process_connection()`](https://nightlies.apache.org/httpd/trunk/doxygen/group__APACHE__CORE__CONNECTION.html#ga69809a83ea209f6140266cd77ab26d0a) hook.

Can you try [compiling mod_antiloris](https://github.com/Deltik/mod_antiloris/blob/e91e7ea8f1ba61697ab3ec80a6091b37820abe81/README.md#compilation-with-cmake) with this patch to skip processing suspended connections to see if the crash stops happening?:

```diff
diff --git a/mod_antiloris.c b/mod_antiloris.c
--- a/mod_antiloris.c	(revision e91e7ea8f1ba61697ab3ec80a6091b37820abe81)
+++ b/mod_antiloris.c	(date 1725643703759)
@@ -279,6 +279,11 @@
 
 /** Our hook at connection processing */
 static int pre_connection(conn_rec *c) {
+    /* Skip suspended connections */
+    if (!c->sbh) {
+        return DECLINED;
+    }
+
     /* Remote IP to be used in checking operations */
     char *remote_ip = c->remote_ip;
 
```

If that fixes mod_antiloris for you, I can make a new release with the fix.

**Comment by multicomunicaec** [comment]
How I get this patch from github, or I edit directly the file?

**Comment by Deltik** [comment]
Directly editing the file would be easiest, since it's just a few lines

**Comment by multicomunicaec** [comment]
The fix it's worked

**Comment by multicomunicaec** [comment]
I don't know if can open this issue again, or is related to this, but i cant compile the module in ubuntu Ubuntu 18.04.3 LTS.

Can you help me with that?

the error show is 
`CMake Error at lib/check/CMakeLists.txt:251 (add_link_options):
  Unknown CMake command "add_link_options".
`

**Comment by Deltik** [comment]
Ah, apparently, the testing library, libcheck, depends on [CMake 3.13](https://github.com/libcheck/check/blob/455005dc29dc6727de7ee36fee4b49a13b39f73f/CMakeLists.txt#L23), but Ubuntu 18.04 comes with CMake 3.10.

You can try downgrading libcheck to version 0.13.0, the last version that reported supporting CMake 3.9 by running this command in your local copy of mod_antiloris:

```shell
git -C "$(git rev-parse --show-toplevel)/lib/check" checkout 0.13.0
```

----

Alternatively, you can upgrade CMake on Ubuntu 18.04 to a newer version with [an official backport](https://askubuntu.com/a/1138455/18979).

**Comment by Deltik** [comment]
@multicomunicaec: Thanks for reporting the bug!

It has now been fixed in [mod_antiloris v0.8.1](https://github.com/Deltik/mod_antiloris/releases/tag/v0.8.1), which you can directly download for x86_64 [here](https://github.com/Deltik/mod_antiloris/releases/download/v0.8.1/mod_antiloris.so).

[The install script](https://github.com/Deltik/mod_antiloris/blob/v0.8.1/README.md#install-script) has also been updated to support Ubuntu 18.04 and compile from source.
